launcher/updatedialog_moc.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+ * updatedialog_moc.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#include "StdInc.h"
+#include "updatedialog_moc.h"
+#include "ui_updatedialog_moc.h"
+
+#include "../lib/CConfigHandler.h"
+#include "../lib/GameConstants.h"
+
+#include <QNetworkAccessManager>
+#include <QNetworkReply>
+#include <QNetworkRequest>
+
+UpdateDialog::UpdateDialog(QWidget *parent, bool calledManually) :
+	QDialog(parent),
+	ui(new Ui::UpdateDialog),
+	calledManually(calledManually)
+{
+	ui->setupUi(this);
+	
+	if(calledManually)
+	{
+		setWindowModality(Qt::ApplicationModal);
+		show();
+	}
+	
+	connect(ui->closeButton, SIGNAL(clicked()), this, SLOT(close()));
+	
+	if(settings["launcher"]["updateOnStartup"].Bool() == true)
+		ui->checkOnStartup->setCheckState(Qt::CheckState::Checked);
+	
+	currentVersion = GameConstants::VCMI_VERSION;
+	
+	setWindowTitle(QString::fromStdString(currentVersion));
+	
+#ifdef VCMI_WINDOWS
+	platformParameter = "windows";
+#elif defined(VCMI_MAC)
+	platformParameter = "macos";
+#elif defined(VCMI_IOS)
+	platformParameter = "ios";
+#elif defined(VCMI_ANDROID)
+	platformParameter = "android";
+#elif defined(VCMI_UNIX)
+	platformParameter = "linux";
+#endif
+	
+	QString url = QString::fromStdString(settings["launcher"]["updateConfigUrl"].String());
+	
+	QScopedPointer<QNetworkAccessManager> manager(new QNetworkAccessManager);
+	
+	QNetworkReply *response = manager->get(QNetworkRequest(QUrl(url)));
+	
+	QObject::connect(response, &QNetworkReply::finished, [&, response]{
+		response->deleteLater();
+		response->manager()->deleteLater();
+		if(response->error() != QNetworkReply::NoError)
+		{
+			ui->versionLabel->setStyleSheet("QLabel { background-color : red; color : black; }");
+			ui->versionLabel->setText("Network error");
+			ui->plainTextEdit->setPlainText(response->errorString());
+			return;
+		}
+		
+		auto byteArray = response->readAll();
+		JsonNode node(byteArray.constData(), byteArray.size());
+		loadFromJson(node);
+		
+	}) && manager.take();
+}
+
+UpdateDialog::~UpdateDialog()
+{
+	delete ui;
+}
+
+void UpdateDialog::showUpdateDialog(bool isManually)
+{
+	UpdateDialog * dialog = new UpdateDialog(nullptr, isManually);
+	
+	dialog->setAttribute(Qt::WA_DeleteOnClose);
+}
+
+void UpdateDialog::on_checkOnStartup_stateChanged(int state)
+{
+	Settings node = settings.write["launcher"]["updateOnStartup"];
+	node->Bool() = (state == 2 ? true : false);
```

Comment:
```suggestion
	node->Bool() = state == 2;
```

Reply:
For me it worse in terms of readability

Reply:
Would it work if the comparison is placed in brackets?

Reply:
it's a matter of taste, there is no improvement

Reply:
Btw might be worth to introduce at least a local constant for 2

Reply:
Probably there should be in QT, I’ll check